### PR TITLE
Releasing updates - better action names, updated documentation, auto create change files

### PR DESCRIPTION
**Pull request by JamesBurnside**
# What
Some improvements to release github actions and documentation after beta release this morning
* Update action names so the steps appear beside each other in the actions tab
* Update to auto create change log files for the merge of the release branch back into main
* Update documentation to better reflect release steps
* update: Do not use cached node modules for npm release to ensure the release is 100% fresh

### CODE REVIEW

File: docs/references/creating-a-release.md [hunk 1]
```diff
@@ -12,20 +12,22 @@ See the [release checklist](../references/release-checklist.md) for tasks that m
...
+1. Ensure the "Before Release" steps of the [release checklist](../references/release-checklist.md) are completed.
+1. Run the "Publish npm packages" GitHub action _off the release branch_.
+    * Enter the tag also, if releasing a new public version the tag name will be `latest`. A beta release would be `next` and an alpha release would be `dev`.
 1. Ensure the action completes successfully then verify on <https://www.npmjs.com/> that the package(s) published successfully.
-1. Post that a new release has happened on the [internal releases Teams channel](https://teams.microsoft.com/l/channel/19%3ae12aa149c0b44318b245ae8c30365880%40thread.skype/ACS%2520Deployment%2520Announcements?groupId=3e9c1fc3-39df-4486-a26a-456d80e80f82&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47).
+1. Complete the post-release verification steps in [Release Checklist](../release-checklist.md).
+1. (If this is a latest release) Complete the PR to merge the release branch back into `main`.
```
Comment (JamesBurnside) [inline docs/references/creating-a-release.md:31]: Add another step/section about releasing new storybook version

File: .github/workflows/npm-release-publish.yml [hunk 2]
```diff
@@ -26,13 +26,6 @@ jobs:
         with:
           node-version: '14.x'
 
-      # Try get node_modules from cache
```
Comment (anjulgarg) [inline .github/workflows/npm-release-publish.yml:29]: This not required anymore?

**Review by anjulgarg — APPROVED** [review]

File: .github/workflows/npm-release-publish.yml (hunk 2 above)
Comment (JamesBurnside) [reply .github/workflows/npm-release-publish.yml:29]: Forgot to mention that in the PR desc, I think we should not be using cached node modules for release and instead making sure the install is 100% fresh

Comment (JamesBurnside) [reply .github/workflows/npm-release-publish.yml:29]: It shouldn't matter and we should be able to use the cached node modules - I'm just being extra paranoid -- happy to put this back in or keep out

**Review by alcail — APPROVED** [review]

**Pull request merged**